cdap-app-fabric/src/main/java/io/cdap/cdap/internal/app/store/AppMetadataStore.java

```diff
@@ -456,7 +478,8 @@ public void updateAppSpec(ApplicationId appId, ApplicationSpecification spec) th
       LOG.trace("Application {} exists in mds with specification {}", appId, GSON.toJson(existing));
     }
     ApplicationMeta updated = new ApplicationMeta(existing.getId(), spec);
-    writeApplicationSerialized(appId.getNamespace(), appId.getApplication(), appId.getVersion(), GSON.toJson(updated));
+    writeApplicationSerialized(appId.getNamespace(), appId.getApplication(), appId.getVersion(), GSON.toJson(updated),
+                               created, updated.getAuthor(), null);
```

Comment:
Why do we need to update the creation time for `updateAppSpec` ? I feel like we should use `existing.getCreated()`

Reply:
This i was not sure of. The creation time is whenever we update the pipeline right - (including app-spec)? However, yes it is not a part of the edit flow.